### PR TITLE
fixed invalid tests.

### PR DESCRIPTION
**Pull request by craiggwilson**
@jyemin fixed up tests.

[Review on Reviewable](https://reviewable.io/reviews/mongodb/specifications/69)

### CODE REVIEW

**Comment by craiggwilson** [comment]
And added a lot more tests as well.

cc: @ajdavis, @bjori, @aherlihy

**Comment by jyemin** [comment]
lgtm.  all passing in the Java driver now.

**Comment by bjori** [comment]
/me admits to not even having noticed there were tests already.

lgtm

File: source/read-write-concern/tests/README.rst [hunk 1]
```diff
@@ -45,22 +45,43 @@ the spec is final.
...
 ------
 
+Connection String
+~~~~~~~~~~~~~~~~~
+
+These tests are designed to excersize the connection string parsing related
+to read concern and write concern.
+
 Each YAML file contains an object with a single ``tests`` key. This key is an
 array of test case objects, each of which have the following keys:
```
Comment (bjori) [inline source/read-write-concern/tests/README.rst]: may.
"each of which may have the following keys".
Not all of them are used all the time

File: source/read-write-concern/tests/README.rst [hunk 2]
```diff
@@ -45,22 +45,43 @@ the spec is final.
...
+for drivers that might substitute default values *during* parsing.
+
+Document
+~~~~~~~~
+
+These tests are designed to ensure compliance with the spec in relation to what should be 
+sent to the server.
+
+Each YAML file contains an object with a single ``tests`` key. This key is an
+array of test case objects, each of which have the following keys:
```
Comment (bjori) [inline source/read-write-concern/tests/README.rst]: Same as above.. not all ``tests have the following keys``, but they probably have most of them

File: source/read-write-concern/tests/connection-string/write-concern.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      }
+    },
+    {
+      "description": "w as an invalid number",
+      "uri": "mongodb://localhost/?w=-2",
+      "valid": false,
+      "warning": null
+    },
+    {
+      "description": "w as a string",
```
Comment (bjori) [inline source/read-write-concern/tests/connection-string/write-concern.json]: I'd update this to say "w as majority" and add at least one test for w=random-gle-tag

**Pull request closed without merging**